src/halmos/sevm.py

```diff
@@ -1168,15 +1174,18 @@ def load(cls, ex: Exec, addr: Any, loc: Word) -> Word:
         num_keys = len(keys)
         size_keys = cls.bitsize(keys)
         if num_keys == 0:
-            return ex.storage[addr][slot][num_keys][size_keys]
+            return ex.storage[addr].mapping[slot][num_keys][size_keys]
         else:
-            if not ex.symbolic:
+            if not ex.storage[addr].symbolic:
                 # generate emptyness axiom for each array index, instead of using quantified formula; see init()
                 ex.path.append(
                     Select(cls.empty(addr, slot, keys), concat(keys)) == ZERO
                 )
             return ex.select(
-                ex.storage[addr][slot][num_keys][size_keys], concat(keys), ex.storages
+                ex.storage[addr].mapping[slot][num_keys][size_keys],
+                concat(keys),
+                ex.storages,
+                ex.storage[addr].symbolic,
             )
```

Comment:
I would avoid the `else` block after the return and decrease nesting here

Reply:
done in #356